i-need-another-coffee/decidim
# Detect the use of spam-bots and ban non compliant users (part 1)

#### :tophat: What? Why?
Implement a way to report users

Like we have a way to flag a contribution for moderation, a similar mechanism can be implemented to flag users and give moderators the ability to block said user. Everyone can participate in this reporting (admin, moderators, users) and flag users based on their harmful behaviour towards the debate or the content they posted on they public profile (avatar, biography, personal website).

    Add a flag to report users on their public profile;
    In the admin, add a column to the participant table that displays the number of time a user was reported and make a sortable column so the admin can see first the ones with most reports and take action (block) if needed;
    Send notification to moderator and admin when a user is reported.

#### :pushpin: Related Issues
Meta Decidim: https://meta.decidim.org/processes/roadmap/f/122/proposals/15628 

#### Testing
1. Open the website, browse the homepage for an event that has an username
2. Navigate the username profile page, and observe the flag button 
3. Click Report (flag icon near nickname), and fill in the data (or login and retry)
4. Navigate to admin Interface in "Participants" -> "Reported Users"
5. Observe the user details (username, reason, justification)
6. Visiting the Participants section, the report count column should be sortable, and shoud display 1 in the corresponding user row
7. Click the unreport functionality, and observe what is happening 
8 Navigate to Participants section, and now the report count should indicate 0


#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.


### :camera: Screenshots
*Please add screenshots of the changes you're proposing*
![Description](URL)
![image](https://user-images.githubusercontent.com/105683/96173553-b5ed3280-0f30-11eb-8d12-85d6524420d3.png)
![image](https://user-images.githubusercontent.com/105683/96173837-1a0ff680-0f31-11eb-8e65-58fae43d6c65.png)
![image](https://user-images.githubusercontent.com/105683/96173900-31e77a80-0f31-11eb-938e-71792bc91ee9.png)
![image](https://user-images.githubusercontent.com/105683/96180531-3d3fa380-0f3b-11eb-8796-f98ace1fe03e.png)
![image](https://user-images.githubusercontent.com/105683/96180968-eb4b4d80-0f3b-11eb-834f-9a1743427121.png)



:hearts: Thank you!
